File: src/main/java/de/schramm/royalbash/tdd/core/Card.java
package de.schramm.royalbash.tdd.core;

import de.schramm.royalbash.tdd.core.Context;
import de.schramm.royalbash.tdd.core.Game;

public interface Card {
    Game invoke(Context context);
    boolean canBePlacedOnSpot();
}

File: src/main/kotlin/de/schramm/royalbash/domain/Card.kt
package de.schramm.royalbash.domain

interface Card {

    val id: String
    val instanceId: String
    val name: String
    val text: String
    val cost: Int
    val image: String

    operator fun invoke(context: InvokationOnSpotContext): Game

    operator fun invoke(context: InvokationOnPlayerContext): Game

    operator fun invoke(context: InvokationOnCreatureContext): Game
}
